rust/processor/src/models/stake_models/current_delegated_voter.rs

```diff
@@ -0,0 +1,370 @@
+                                        delegation_pool_address:
+                                            delegation_pool_address_from_query_result.clone(),
+                                        voter: Some(voter.clone()),
+                                        pending_voter: Some(pending_voter.clone()),
+                                        last_transaction_timestamp: txn_timestamp,
+                                        last_transaction_version: txn_version,
+                                        table_handle: table_handle.clone(),
+                                    };
+                                    current_delegated_voter_map.insert(
+                                        (
+                                            delegation_pool_address_from_query_result.clone(),
+                                            delegator_address.clone(),
+                                        ),
+                                        current_delegated_voter,
+                                    );
+                                    vote_delegation_handle_to_pool_address.insert(
+                                        table_handle.clone(),
+                                        delegation_pool_address_from_query_result.clone(),
+                                    );
+                                    pk_to_vote_delegation.insert(
+                                        (
+                                            delegation_pool_address_from_query_result.clone(),
+                                            delegator_address.clone(),
+                                        ),
+                                        (
+                                            Some(voter.to_string()),
+                                            Some(pending_voter.to_string()),
+                                            table_handle.clone(),
+                                        ),
+                                    );
+                                },
+                                Err(_) => {
+                                    tracing::error!(
+                                        transaction_version = txn_version,
+                                        lookup_key = &table_handle,
+                                        "Failed to get delegation pool address from vote delegation write table handle. You probably should backfill db.",
+                                    );
+                                },
+                            }
+                        }
+                    }
+                }
+            }
+
+            // this write resource indexing is to get active pool to DelegatorPoolBalanceMetadata mapping
+            // as a helper for us to get delegation pool address and delegator address from DelegationPool
+            // with the assumption that when delegator can delegate votes, they must have already staked
+            for wsc in &transaction.info.as_ref().unwrap().changes {
```

Comment:
@bowenyang007 also mentioned that we can probably reuse some of the functions from `delegator_balances.rs`